pkg/broker/handler/pool/fanout/pool.go

```diff
@@ -33,23 +33,31 @@ import (
 	"github.com/google/knative-gcp/pkg/broker/handler/processors/deliver"
 	"github.com/google/knative-gcp/pkg/broker/handler/processors/fanout"
 	"github.com/google/knative-gcp/pkg/broker/handler/processors/filter"
+	"github.com/google/knative-gcp/pkg/utils"
 )
 
 // SyncPool is the sync pool for fanout handlers.
 // For each broker in the config, it will attempt to create a handler.
 // It will also stop/delete the handler if the corresponding broker is deleted
 // in the config.
 type SyncPool struct {
-	options *pool.Options
-	targets config.ReadonlyTargets
-	pool    sync.Map
+	options   *pool.Options
+	targets   config.ReadonlyTargets
+	pool      sync.Map
+	projectID string
 }
 
 // StartSyncPool starts the sync pool.
 func StartSyncPool(ctx context.Context, targets config.ReadonlyTargets, opts ...pool.Option) (*SyncPool, error) {
+	options := pool.NewOptions(opts...)
+	projectID, err := utils.ProjectID(options.ProjectID)
```

Comment:
I'm thinking maybe do that in main.go and pass in as an option?

Reply:
We already do that in `main`: https://github.com/google/knative-gcp/blob/master/cmd/broker/fanout/main.go#L184

This change is to handle the case where the PROJECT_ID env var is absent.

Reply:
Or do you mean `main` should call `utils.ProjectID` and pass it as option?

I think we should do it here because projectID is optional. The pool should be able to find a good default if option is absent. Otherwise we should make project id mandatory for the pool.

Reply:
Oh I misunderstood what `utils.ProjectID` does. You're right. An alternative is to reuse `options.ProjectID` instead of adding a new struct field.

Reply:
Using a struct field because `options.ProjectID` can be empty. We can of course override it with the default but I think let's keep them separate to avoid surprises. So `options` always contains original values that the caller provides.

Reply:
`options.ProjectID` being empty means no project ID is set and it's ok to assign default value to it. In fact, I would rather it be in `options.go` but that requires `NewOptions` to return error (which I now think is a good idea because we can also add validation there). You can leave that to me if you just want to keep this change small.